### ISSUE TITLE
GPU BinaryCrossEntropy Failure

### ISSUE THREAD

**Issue by shtoshni** [liufly/refreader#1]
Hey Fei,

My jobs on TitanX are crashing due to an error saying that the inputs are not between 0 & 1. 
The model does train for about 10 epochs before crashing. Also, how many epochs do you train it for? The performance seems to saturate within the first few epochs for me.

**Comment by liufly** [comment]
@shtoshni92 
I have only tested the code on CPUs (as you may have noticed that I have commented out a few lines regarding CUDA in `train.py` and `trainer.py`).
That said, I used to be able to train the model on a single GPU for much longer than 10 epochs.
What is the error message?

And yes, model performance seems to saturate within the first few epochs.

**Comment by shtoshni** [comment]
Yeah, I uncommented those lines to train them faster.

Here's the exact error as I see:
> /opt/conda/conda-bld/pytorch_1556653183467/work/aten/src/THCUNN/BCECriterion.cu:57: void bce_updateOutput_no_reduce_functor<Dtype, Acctype>::operator()(const Dtype *, const Dtype *, Dtype *) [with Dtype = float, Acctype = float]: block: [961,0,0], thread: [448,0,0] Assertion `*input >= 0. && *input <= 1.` failed.

Interesting, yeah for me the jobs kept failing around 10 epochs. Also, I was trying to train the model with more number of memory cells, so that might have something to do with it. 

Yeah I don't find the performance to improve beyond 5 epochs.

**Comment by liufly** [comment]
We did make the last minute env switch (from PyTorch 0.4 to >= 1.0) to make the code compatible with the more recent, stable release version of PyTorch.
As far as I remember, there was little change in the code itself.

I am pretty sure that we were able to train the model on a single GPU for much longer than 10 epochs with PyTorch 0.4 (I did try 4 memory cells as well and it too worked fine without crashing).
Perhaps the difference in PyTorch version is what's causing the problem?
Sorry I don't have access to a GPU at the moment to test it.
